lib/spack/docs/getting_started.rst

```diff
@@ -600,6 +600,61 @@ flags to the ``icc`` command:
            spec: intel@15.0.24.4.9.3
 
 
+^^^^^^^^^^
+SYCL/DPC++
+^^^^^^^^^^
+
+The SYCL/DPC++ compiler is Intel and Codeplay's `staging area compiler <https://github.com/intel/llvm>`_ for upcoming Clang/LLVM SYCL support.
+They are usually distributed with `oneAPI <https://software.intel.com/content/www/us/en/develop/articles/oneapi-repo-instructions.html>`_ and compilers are activated with ``source /opt/intel/inteloneapi/setvars.sh`` on Linux.
+
+One can record these activation steps by comparing set environment variables:
+
+.. code-block:: console
+
+   env > env.txt
+   set +e
+   source /opt/intel/inteloneapi/setvars.sh
+   env > env_extras.txt
+   diff env.txt env_extras.txt
+
+All newly set environment hints will then be shown in this diff.
+A typical entry in ``compilers.yaml`` looks like this:
+
+.. code-block:: yaml
+
+   # DPC++ 2021.1-beta07 (2020.5.0.0604)
+   - compiler:
+       spec: clang@11.0.0
+       paths:
+         cc: /opt/intel/inteloneapi/compiler/latest/linux/bin/clang
+         cxx: /opt/intel/inteloneapi/compiler/latest/linux/bin/dpcpp
+         f77:
+         fc:
+       flags: {}
+       operating_system: ubuntu18.04
+       target: x86_64
+       modules: []
+       environment:
+         set:
+           OCL_ICD_FILENAMES: libintelocl_emu.so:libalteracl.so:/opt/intel/inteloneapi/compiler/latest/linux/lib/x64/libintelocl.so
+           ACL_BOARD_VENDOR_PATH: /opt/Intel/OpenCLFPGA/oneAPI/Boards
+           FI_PROVIDER_PATH: /opt/intel/inteloneapi/mpi/2021.1-beta07//libfabric/lib/prov
+           CLASSPATH: /opt/intel/inteloneapi/mpi/2021.1-beta07//lib/mpi.jar
+           INTELFPGAOCLSDKROOT: /opt/intel/inteloneapi/compiler/2021.1-beta07/linux/lib/oclfpga
+           NLSPATH: /opt/intel/inteloneapi/mkl/2021.1-beta07/lib/intel64/locale/%l_%t/%N
+           DPCPP_ROOT: /opt/intel/inteloneapi
+           ONEAPI_ROOT: /opt/intel/inteloneapi
+           MKLROOT: /opt/intel/inteloneapi/mkl/2021.1-beta07
+           TBBROOT: /opt/intel/inteloneapi/tbb/2021.1-beta07/env/..
+           I_MPI_ROOT: /opt/intel/inteloneapi/mpi/2021.1-beta07
```

Comment:
Some of these environment variables don't properly belong as part of the compiler. The compiler should NOT know about I_MPI or MKL unless the user also requested one of those as an external package.

Reply:
Yeah, confused me as well... guess that's because I installed more basic packages. Will remove.